### DISCUSSION TITLE
Unable to reload page via cogctl

### DISCUSSION THREAD

**Discussion by WoeMat** [Igalia/cog#428] [Q&A]
I'm using cog with an Raspberry CM3 to display a GUI. The GUI itself is a single page web application and every time I am changing something on the website and loading it onto the CM3 the changes don't appear on the screen.
I think it is all about the caching of the browser so I tryed to start cog like this.

`cog --platform=fdo  --enable-offline-web-application-cache=false  --enable-page-cache=false http://127.0.0.1  `

but still changes are not displayed. 
I alos tried to reload by using cogctl but than this error occured.

```
(cogctl:6293): GLib-GIO-CRITICAL **: 09:23:05.575: g_dbus_connection_call_sync_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Cannot autolaunch D-Bus without X11 $DISPLAY
```
Is the only way to reload the page cogctl or can I do something else to see my changes be displayed?

In this Setup I am using 
Weston 9.0.0
cog 0.8.1 (WPE WebKit 2.30.6)

Thanks

**Comment by aperezdc** [comment]
@WoeMat Is the environment variable `DBUS_SESSION_BUS_ADDRESS` defined to a suitable value? You can check which value the `cog` process is using with this command:

```sh
xargs -L1 -0 echo < /proc/$(pidof cog)/environ | grep '^DBUS_'
```

Then you should be able to use:

```sh
export DBUS_SESSION_BUS_ADDRESS=<value found above>
cogctl reload
```

If you have built Cog with the option to expose the D-Bus interface in the system bus, you need to pass `--system` to the command:

```sh
cogctl --system reload
```

**Reply by WoeMat** [reply]
Ok, that seems to be the Problem. There is no DBUS Address in 
`/proc/$(pidof cog)/environ`

**Comment by WoeMat** [comment]
@aperezdc When I try your second command I  get this:
```
# cogctl --system reload
GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Name "com.igalia.Cog" does not exist
```

**Reply by AminSahebi** [reply]
Hi @aperezdc 

I am getting the above error in terms of D-Bus error but with the following command "cogctl --system reload" and I guess this is related to the build procedure, my build environment is yocto and my target is an embedded device, is it necessary to build the cog with a specific argument to add this option? I think there is something missing here with the cog build procedure and D-Bus! 
bw I get nothing by issuing the following command: 

xargs -L1 -0 echo < /proc/$(pidof cog)/environ | grep '^DBUS_'

I appreciate any advice, 
Thanks

**Reply by aperezdc** [reply]
If you are looking into using the system bus, you need to build with the [cog_dbus_control option](https://github.com/Igalia/cog/blob/master/meson_options.txt#L53) configured accordingly. Note that it is strongly recommended to run Cog as a regular, unprivileged user, so you may want to also set `cog_dbus_system_owner` option to allow some user other than `root` to own the well-known name in the system bus:

```
meson setup builddir -Dcog_dbus_control=system -Dcog_dbus_system_owner=coguser ...
```

For people using Yocto, the [build recipe](https://github.com/Igalia/meta-webkit/blob/main/recipes-browser/cog/cog-meson.inc#L3) has a `PACKAGECONFIG` option that can be set, so I believe adding something like this in your `local.conf` file should work:

```
PACKAGECONFIG:pn-cog:append = " dbus"
COG_DBUS_OWN_USER = "weston"
```

**Reply by AminSahebi** [reply]
Thanks Dear Adrian,

I have been struggling a lot with the yocto build and the system bus to make the Cog browser work within the system bus session, after all, I faced the following error:

>  (cogctl:3570): GLib-GIO-CRITICAL **: 15:07:24.635: g_dbus_connection_call_sync_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Error spawning command line ?dbus-launch --autolaunch=ec5029f6d7284741bdcaf7c8d11e3b23 --binary-syntax --close-stderr?: Child process exited with code 1


I executed the Cog and Cogctl as follows. (Apparently, the control interface through dbus doesn't work which is strange.)

` XDG_RUNTIME_DIR=/tmp DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/dbus/system_bus_socket G_MESSAGES_DEBUG=Cog-DRM cog --platform=drm --platform-params=renderer=gles --gapplication-app-id=com.example.Cog1 www.yahoo.com `

I opened the dbug messages within `G_MESSAGES_DEBUG=Cog-DRM` and named an interface for each run like `--gapplication-app-id=com.example.Cog1`

and trying to control via cogctl for this `--appid=com.example.Cog1` 

`XDG_RUNTIME_DIR=/tmp DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/dbus/system_bus_socket cogctl -A com.example.Cog1 quit`

And finally, the following error happens: 

>  (cogctl:3570): GLib-GIO-CRITICAL **: 15:07:24.635: g_dbus_connection_call_sync_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Error spawning command line ?dbus-launch --autolaunch=ec5029f6d7284741bdcaf7c8d11e3b23 --binary-syntax --close-stderr?: Child process exited with code 1

**PS1**: I don't see any dbus message on systemctl status (or journalctl -u ) logs when I run Cog as a service, 

>Acquired D-Bus well-known name com.igalia.Cog 

**PS2**: The Yocto build procedure creates the correct "com.igalia.Cog.conf" file and is located in /etc/dbus-1/system.d on my target.

my local.conf file is also attached,

Any advice is highly appreciated,
[local.conf.txt](https://github.com/Igalia/cog/files/11330740/local.conf.txt)

**Reply by AminSahebi** [reply]
I solved this issue, It was mainly some mistake on yocto recipe configuration.
